### DISCUSSION TITLE
[Request] Unit type filters

### DISCUSSION THREAD

**Discussion by RexTheWho** [diesel-modding/BeardLib-Editor#616] [Ideas]
In the unit file there's a type value that seems to be used for searching groups of units from a DLC.
`<unit type="gen" slot="1">`

Being able to use this somehow for spawning or searching for units would be handy.

Context in overkills editor, a short ID filter section.
![image](https://user-images.githubusercontent.com/26837908/86529019-ca587280-bedf-11ea-8f82-883f5c8ea792.png)

**Comment by Luffyyy** [comment]
- [x] Searching units
- [ ] Spawning units

Spawning units is a little more tricky, it's pretty slow to parse the XML of their unit file and using the PackageManager:unit_data is also sort of slow, but has a tendency of crashing (you'd crash if you'll try to spawn the problematic unit, but at least you'll know which unit that was vs just opening the list and crashing), I'd like to stop using it at some point.
